In [1]:
report_card_dictionary = {"student_names": ["Vanshika","Vivvan","Mahir","Kareena","Mitansh"],
                         "physics_grades": [34,66,77,98,33],
                         "chemistry_grades":[56,88,44,77,88],
                         "math_grades":[44,55,88,33,56],
                          "biology_grades":[58,98,67,45,65]
                         } 
report_card_dictionary 

{'student_names': ['Vanshika', 'Vivvan', 'Mahir', 'Kareena', 'Mitansh'],
 'physics_grades': [34, 66, 77, 98, 33],
 'chemistry_grades': [56, 88, 44, 77, 88],
 'math_grades': [44, 55, 88, 33, 56],
 'biology_grades': [58, 98, 67, 45, 65]}

In [2]:
# Setup
import pandas as pd 

In [3]:
#Making the report card dictionary into a dataframe
report_card_df = pd.DataFrame(report_card_dictionary )
report_card_df 

student_names  physics_grades  chemistry_grades  math_grades  biology_grades
0      Vanshika              34                56           44              58
1        Vivvan              66                88           55              98
2         Mahir              77                44           88              67
3       Kareena              98                77           33              45
4       Mitansh              33                88           56              65

In [4]:
# First 3 rows in the report card dataframe
report_card_df.head(3) 

student_names  physics_grades  chemistry_grades  math_grades  biology_grades
0      Vanshika              34                56           44              58
1        Vivvan              66                88           55              98
2         Mahir              77                44           88              67

In [5]:
# 3 random rows in the report card dataframe
report_card_df.sample(3)  

student_names  physics_grades  chemistry_grades  math_grades  biology_grades
4       Mitansh              33                88           56              65
2         Mahir              77                44           88              67
3       Kareena              98                77           33              45

In [6]:
# Describing all the numeric colums in the report card dataframe
report_card_df.describe() 

physics_grades  chemistry_grades  math_grades  biology_grades
count        5.000000          5.000000     5.000000        5.000000
mean        61.600000         70.600000    55.200000       66.600000
std         28.112275         19.793938    20.583974       19.552493
min         33.000000         44.000000    33.000000       45.000000
25%         34.000000         56.000000    44.000000       58.000000
50%         66.000000         77.000000    55.000000       65.000000
75%         77.000000         88.000000    56.000000       67.000000
max         98.000000         88.000000    88.000000       98.000000

# Insights 

- what is the average chemistry grade? 70.6
- what is the average biology grade? 66.6
- lowest math grade? 33
- highest physics grade? 98
- how many scores are present in the math column? 5

In [7]:
# Make a total grade column

report_card_df["total_grades"] = report_card_df["physics_grades"] +report_card_df["math_grades"] +report_card_df["chemistry_grades"]+report_card_df["biology_grades"]

#Printed the report card dataframe to see the added column 
report_card_df   

student_names  physics_grades  chemistry_grades  math_grades  \
0      Vanshika              34                56           44   
1        Vivvan              66                88           55   
2         Mahir              77                44           88   
3       Kareena              98                77           33   
4       Mitansh              33                88           56   

   biology_grades  total_grades  
0              58           192  
1              98           307  
2              67           276  
3              45           253  
4              65           242

# Challenge : Make a column called average grade 

In [9]:
report_card_df["ave_grade"] = report_card_df["total_grades"]/4
report_card_df 

student_names  physics_grades  chemistry_grades  math_grades  \
0      Vanshika              34                56           44   
1        Vivvan              66                88           55   
2         Mahir              77                44           88   
3       Kareena              98                77           33   
4       Mitansh              33                88           56   

   biology_grades  total_grades  ave_grade  
0              58           192      48.00  
1              98           307      76.75  
2              67           276      69.00  
3              45           253      63.25  
4              65           242      60.50

In [10]:
report_card_df.describe() 

physics_grades  chemistry_grades  math_grades  biology_grades  \
count        5.000000          5.000000     5.000000        5.000000   
mean        61.600000         70.600000    55.200000       66.600000   
std         28.112275         19.793938    20.583974       19.552493   
min         33.000000         44.000000    33.000000       45.000000   
25%         34.000000         56.000000    44.000000       58.000000   
50%         66.000000         77.000000    55.000000       65.000000   
75%         77.000000         88.000000    56.000000       67.000000   
max         98.000000         88.000000    88.000000       98.000000   

       total_grades  ave_grade  
count      5.000000   5.000000  
mean     254.000000  63.500000  
std       42.667318  10.666829  
min      192.000000  48.000000  
25%      242.000000  60.500000  
50%      253.000000  63.250000  
75%      276.000000  69.000000  
max      307.000000  76.750000

In [12]:
report_card_insights_df = report_card_df.describe().transpose()
report_card_insights_df  

count   mean        std    min    25%     50%    75%     max
physics_grades      5.0   61.6  28.112275   33.0   34.0   66.00   77.0   98.00
chemistry_grades    5.0   70.6  19.793938   44.0   56.0   77.00   88.0   88.00
math_grades         5.0   55.2  20.583974   33.0   44.0   55.00   56.0   88.00
biology_grades      5.0   66.6  19.552493   45.0   58.0   65.00   67.0   98.00
total_grades        5.0  254.0  42.667318  192.0  242.0  253.00  276.0  307.00
ave_grade           5.0   63.5  10.666829   48.0   60.5   63.25   69.0   76.75

In [13]:
report_card_insights_df.to_csv("report card insights.csv") 